### PR TITLE
Hack in an updated GCC13 container.

### PR DESCRIPTION
**Pull request by wmaxey**

### CODE REVIEW

File: features/src/gcc/install.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ GCC_VERSION="${VERSION:-${GCC_VERSION_DEFAULT}}";
 
 if [[ "${GCC_VERSION}" != "${GCC_VERSION_DEFAULT}" ]]; then
     apt-add-repository -y ppa:ubuntu-toolchain-r/test;
+    apt-add-repository -y ppa:ubuntu-toolchain-r/ppa;
```
Comment (wmaxey) [inline features/src/gcc/install.sh:28]: This enables installing GCC 13.2 as opposed to GCC 13.1 in the test PPA.

**Comment by wmaxey** [comment]
I rebased to be up to date with the target branch. I think that the build failure last seen were not related my to change, and seemed more like timeouts with various repos.

**Comment by jrhemstad** [comment]
I figured out why the gcc13 jobs are failing.

I figured out why your gcc13 changes are failing.

It's ultimately because there is no Ubuntu 23.04 CTK package here: https://developer.download.nvidia.com/compute/cuda/repos/

So that breaks this: https://github.com/rapidsai/devcontainers/blob/99da203868d529cae7c9364f55cdd15aa77ba052/features/src/cuda/install.sh#L25-L52

File: features/src/cuda/install.sh [hunk 2]
```diff
@@ -36,6 +36,10 @@ get_cuda_deb() {
     | grep -P "^Filename: \./${2}(.*)\.deb$"      \
     | sort -Vr | head -n1 | cut -d' ' -f2         \
     )";
+    if [ -z "$deb" ]; then
+        echo "Error: No matching .deb found for '${1}' and '${2}'" >&2
+        return 1
+    fi
```
Comment (trxcllnt) [inline features/src/cuda/install.sh:42]: This won't work for Ubuntu 23.04, we'll need some way to use the CUDA Ubuntu 22.04 apt repo.

**Pull request merged**